src/main/scala/io/iohk/ethereum/keystore/KeyStore.scala

```diff
@@ -99,6 +113,15 @@ class KeyStoreImpl(keyStoreDir: String, secureRandom: SecureRandom) extends KeyS
     }
   }
 
+  private def overwrite(name: String, encKey: EncryptedKey): Either[KeyStoreError, Unit] = {
+    val json = EncryptedKeyJsonCodec.toJson(encKey)
+    val path = Paths.get(keyStoreDir, name)
+    Try {
+      Files.write(path, json.getBytes(StandardCharsets.UTF_8))
+      ()
+    }.toEither.left.map(ioError)
```

Comment:
if an error happens, can the previous key get lost?

Reply:
If the error somehow occurs during write (some bytes written, but not all - probably also depends on FS) then I guess it's possible.

Reply:
Of course trying to revert in case of error may also result in an error. The baseline is that storing these accounts does not relieve the user of responsibility to know his private key (or the passphrase to generate it). So I don't consider the unlikely event of losing the key this way to be a terrible thing.

Unless you see an easy solution for this, we can either schedule a follow-up task or simply ignore this issue (and I'm more inclined towards the latter).

Reply:
I don't see an easy solution but I would be really sad if a client loses my wallet because I changed the passphrase. What we can do is to let daedalus guys know so they can put a big warning sign.